src/api_rework.jl

```diff
@@ -0,0 +1,120 @@
+# I am putting everything in here for now. The contents of this file should be moved around in the future.
+"""
+    findpeaks(x) -> NamedTuple
+
+Find the peaks in a vector `x`. 
+A `NamedTuple` is returned with the original vector 
+in the field `data`, and the indices of the peaks 
+in the field `inds`.
+
+This function serves as the entry-point for other 
+functions such as `peakproms!` and `peakwidths!`
+"""
+function findpeaks(x::AbstractVector)
+    data = x
```

Comment:
```suggestion
```

Remember that Julia doesn't copy on assignment, so this line isn't necessary (and also not used on line 16 in the return statement).

Reply:
Goot catch, this is a brainfart on my behalf. I changed it to a one-liner: 
```julia
findpeaks(x::AbstractVector) = (data=x, inds=findmaxima(x)[1])
```